ament_index_python/ament_index_python/cli.py

```diff
@@ -71,7 +71,7 @@ def resource_name_completer(prefix, parsed_args, **kwargs):
     resource_type = getattr(parsed_args, 'resource_type', None)
     if not resource_type:
         return []
-    return get_resources(resource_type).keys()
+    return (t for t in list(get_resources(resource_type).keys()) if t.startswith(prefix))
```

Comment:
nit: you don't need a convertion to `list`:
```suggestion
    return (t for t in get_resources(resource_type).keys() if t.startswith(prefix))
```